docs/dev/setup/kubernetes.rst

```diff
@@ -0,0 +1,353 @@
+      k3d cluster list 
+      kubectl get nodes 
+      k3d kubeconfig get ${env:CLUSTER_NAME} > $env:KUBECONFIG_FILE 
+      $env:KUBECONFIG=($env:KUBECONFIG_FILE)
+
+3. Install cert-manager
+   
+   cert-manager is used to add certificates and certificate issuers as resource types in Kubernetes clusters. 
+   It simplifies the process of obtaining, renewing and using those certificates.
+   It can issue certificates from a variety of supported sources, i.e. Let’s Encrypt, HashiCorp Vault, Venafi.
+   
+   In our case it will issue self-signed certificates to our Kubernetes deployments to secure the communication between the different deployments.
+
+   Before the installation, you need to add the Jetstack repository and update the local Helm chart repository cache.
+   cert-manager has to be installed in a separate namespace called ``cert-manager`` so one should be created as well. After the installation you can check the status of the installation.
+
+   ::
+
+      helm repo add jetstack https://charts.jetstack.io 
+      helm repo update
+      kubectl create namespace cert-manager
+      helm install cert-manager jetstack/cert-manager --namespace cert-manager --version v1.0.4 --set installCRDs=true --wait 
+      kubectl -n cert-manager rollout status deploy/cert-manager
+
+4. Install Rancher
+
+   Rancher is a Kubernetes management tool which gives you the opportunity to create and manage Kubernetes deployments more easily than with the CLI tools.
+   
+   You can install Rancher using helm - the package manager for Kubernetes. It has to be installed in a namespace called ``cattle-system`` and we should create such a namespace before the installation itself.
+   During the installation we set the namespace and the hostname which Ranchen will be accessible on.
+   Then we can check the installation status. 
+
+   **For MacOS/Linux**:
+
+   ::
+
+      helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
+      helm repo update
+      kubectl create namespace cattle-system
+      helm install rancher rancher-latest/rancher --namespace cattle-system --set hostname=$RANCHER_SERVER_HOSTNAME --wait 
+      kubectl -n cattle-system rollout status deploy/rancher
+
+   **For Windows**:
+   
+   ::
+
+      helm repo add rancher-latest https://releases.rancher.com/server-charts/latest
+      helm repo update
+      kubectl create namespace cattle-system
+      helm install rancher rancher-latest/rancher --namespace cattle-system --set hostname=${env:RANCHER_SERVER_HOSTNAME} --wait 
+      kubectl -n cattle-system rollout status deploy/rancher
+
+5. Open Rancher and update the password
+
+Open Rancher on `<https://rancher.localhost/>`__.
+
+You will be notified that the connection is not private. The reason for that is that the Rancher deployment uses a self-signed certificate by an unknown authority 'dynamiclistener-ca'. 
```

Comment:
Maybe add the hint, that if it does not work with chrome, use firefox as alternative browser. I only managed to open `rancher.localhost` in firefox.